index.js

```diff
@@ -19,10 +28,76 @@ module.exports = {
         return converter.convert(data, options, cb);
       });
     }
-    return cb(null, {
-      result: false,
-      reason: 'input type:' + input.type + ' is not valid'
-    });
+    else if (input.type === 'folder') {
+      let filesPathArray = input.data,
+        convertedSpecs = [],
+        rootFiles = parse.getRootFiles(filesPathArray);
+
+      async.eachSeries(rootFiles, (rootFile, callback) => {
+        parse
+        // will merge all the files in the folder
+          .mergeFiles(rootFile, OasResolverOptions)
+          .then((spec) => {
+            converter.convert(spec, options, (err, result) => {
+              if (err) {
+                return callback(null, {
```

Comment:
What happens to this failure? Will it be visible in any form in the final return value?